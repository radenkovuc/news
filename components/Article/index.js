import styled from 'styled-components'
import React from 'react'
import Router from 'next/router'
import {useTranslation} from 'next-i18next'

const ContentContainer = styled.div`
  padding: 10px;

  @media only screen and (min-width: 768px) {
    padding: 30px;
  }
`

const Title = styled.div`
  font-size: 22px;
  font-weight: 600;
  font-family: 'Nunito Sans black', sans-serif;
  text-align: start;
  padding-bottom: 10px;
`

const Image = styled.img`
  width: 100%;
`

const Content = styled.div`
  font-size: 18px;
  font-family: 'Nunito Sans black', sans-serif;
  text-align: start;
  padding-top: 10px;
`
const Back = styled.div`
  display: flex;
  justify-content: flex-start;
  padding-top: 20px;
  cursor: pointer;
`
const BackText = styled.div`
  font-family: 'Nunito Sans black', sans-serif;
  font-size: 22px;
  padding-left: 5px;
`
const BackIcon = styled.div`
  width: 18px;
  height: 24px;
  background: url('/static/images/prev.svg') no-repeat center/cover;
`

type Props = {
  article: Object
}

const Article = ({article}: Props) => {
  const {t} = useTranslation()

  return (
    <ContentContainer>
      <Title>{article.title}</Title>
      <Image src={article.urlToImage} />
      <Content>{article.content}</Content>
      <Back onClick={() => Router.back()}>
        <BackIcon />
        <BackText>{t('BACK_TO_LIST')}</BackText>
      </Back>
    </ContentContainer>
  )
}

Article.defaultProps = {
  article: {}
}

export default Article
