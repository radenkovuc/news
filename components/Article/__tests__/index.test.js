import React from 'react'
import Article from '../'
import renderer from 'react-test-renderer'

describe('<Article />', () => {
  it('match snapshot', () => {
    const tree = renderer.create(<Article />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
