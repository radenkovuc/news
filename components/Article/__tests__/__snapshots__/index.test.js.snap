// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Article /> match snapshot 1`] = `
<div
  className="ContentContainer-sc-1f0ayl8-0 cfjAmd"
>
  <div
    className="Title-sc-1f0ayl8-1 dwzDBp"
  />
  <img
    className="Image-sc-1f0ayl8-2 ilsyit"
  />
  <div
    className="Content-sc-1f0ayl8-3 fnPcPG"
  />
  <div
    className="Back-sc-1f0ayl8-4 ehdbsJ"
    onClick={[Function]}
  >
    <div
      className="BackIcon-sc-1f0ayl8-6 bFZjGm"
    />
    <div
      className="BackText-sc-1f0ayl8-5 ffQJYe"
    >
      BACK_TO_LIST
    </div>
  </div>
</div>
`;
